src/renderer/gdi/paint.cpp

```diff
@@ -284,61 +284,68 @@ using namespace Microsoft::Console::Render;
 // See: Win7: 390673, 447839 and then superseded by http://osgvsowi/638274 when FE/non-FE rendering condensed.
 //#define CONSOLE_EXTTEXTOUT_FLAGS ETO_OPAQUE | ETO_CLIPPED
 //#define MAX_POLY_LINES 80
-[[nodiscard]] HRESULT GdiEngine::PaintBufferLine(std::basic_string_view<Cluster> const clusters,
+[[nodiscard]] HRESULT GdiEngine::PaintBufferLine(_Inout_ RenderClusterIterator& clusterIter,
                                                  const COORD coord,
                                                  const bool trimLeft) noexcept
 {
     try
     {
-        const auto cchLine = clusters.size();
-
-        // Exit early if there are no lines to draw.
-        RETURN_HR_IF(S_OK, 0 == cchLine);
-
+        const size_t preallocateSize = (_rcInvalid.right - _rcInvalid.left) / 8;

```

Comment:
I found that `8` is the magic number needed here to get the correct size? Any GDI+ reference that explains this?